### PR TITLE
Test Mechanism of Joke App

### PR DESCRIPTION
**Pull request by OnurSefa**
I have offered a test mechanism for Joke App inside the practice_app directory which checks the capability of the app to handle the viewing operations and API call operations. It checks the following API's: My first API for getting user information. It shows the user name and the liked jokes of the user. Second API is for adding a user to db. It addes a new document to mongoDB collection and creates an empty list of liked jokes. Third API is for updating the favorite jokes of a user. It adds one by one the favored jokes. Last one is for deleting a user from the db. Actually our application does not allow to enable a user to delete itself but in test mechanism we should remove the addition operation. With this branch I also modified the return types of the API helper functions regarding to the json format.

### CODE REVIEW

**Review by fleafclover — COMMENTED** [review]
everthing is looking great. Although some of the tests seems like not testing the return value, I think for that functions checking return values are impossible. Therefore we can merge that branch.

**Review by gulsahc — APPROVED** [review]

**Review by fleafclover — APPROVED** [review]

File: practice_app/tests/test.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        status = response.status_code
+        self.assertEqual(200, status)
+        self.assertFalse(not equality_check(response.json, mock_data))
+
+    # test for adding a user
+    def test_api_add_user(self):
+        tester = app.test_client(self)
+        response = tester.post("/joke/add_user/{}".format(will_deleted_mock_data['name']))
+        status = response.status_code
+        self.assertEqual(200, status)
```
Comment (safaaltunel) [inline practice_app/tests/test.py:106]: Your api should return 201 for post requests. You can change your api and then in this test you can compare your status code with 201

File: practice_app/tests/test.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        self.assertFalse(not equality_check(response.json, mock_data))
+
+    # test for adding a user
+    def test_api_add_user(self):
+        tester = app.test_client(self)
+        response = tester.post("/joke/add_user/{}".format(will_deleted_mock_data['name']))
+        status = response.status_code
+        self.assertEqual(201, status)
```
Comment (OnurSefa) [inline practice_app/tests/test.py:106]: I have changed the return status code of the post API which is implemented in the helper_functions.py. Regarding to this change I also changed the validity condition of the related API.

Comment (OnurSefa) [reply practice_app/tests/test.py:106]: Related change request mentioned from @safaaltunel

**Pull request merged**